### PR TITLE
*: support paging protocol on unistore

### LINKED ISSUE #35243: implement coprocessor paging protocol on unistore
## Enhancement

After https://github.com/pingcap/tidb/pull/35114, chunk RPC encoding on unistore is no possible.
So I want to implement the coprocessor paging protocol.

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #35243 #35242 

Problem Summary:

I'm try to support coprocessor paging protocol on unistore, and found those issues:

- [ ] There are several kinds of coprocessor request, including DAG and others. `admin check table/index` `analyze` are not using DAG so they don't support paging
- [ ] Find bug https://github.com/pingcap/tidb/issues/35242 that for partition table on dynamic pruning mode, the key ranges is not sorted. So rebuilding cop tasks and calculate remain key ranges get wrong result.
- [ ] cop cache for paging is not well supported.
- [ ] When enable/disable paging, the explain result (related to IndexLookup) is different, so some test cases need to be updated.
- [ ] QueryFeedback is broken when coprocessor paging is enabled.
- [ ] mocktikv doesn't support paging, and some test cases are still using mocktikv, so if we enable paging by default, those test cases needs update
- [ ] Read from `INFORMATION_SCHEMA.CLUSTER_XXX` table use coprocessor request to query tidb instances, and tidb coprocessor doesn't support paging

### What is changed and how it works?

You can ignore all the xxx_test.go files and just review cop_handler.go / mpp_exec.go 
The actual changes are there.

In brief, I update the paging in tablescan and indexscan's `next()` function to keep the paging key range update to date.
And in the out most loop, check if the chunk pages are enough to break the `next()` looping.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)

I make session variable `tidb_enable_paging` default value to `On` and run all the unit tests to check everything go work well.

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @XuHuaiyu @wshwsh12

**Comment by tiancaiamao** [comment]
PTAL @XuHuaiyu @wshwsh12

**Review by wshwsh12 — APPROVED** [review]
LGTM

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -82,6 +82,17 @@ func (c *CopClient) Send(ctx context.Context, req *kv.Request, variables interfa
...
+	failpoint.Inject("DisablePaging", func(_ failpoint.Value) {
+		req.Paging = false
+	})
+	if req.StoreType == kv.TiDB {
+		// coprocessor on TiDB doesn't support paging
+		req.Paging = false
+	}
+	if req.Tp != kv.ReqTypeDAG {
+		// coprocessor request but type is not DAG
+		req.Paging = false
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply store/copr/coprocessor.go:93]: paging is a property of the kv request,
kv request can be one of the following kinds:

```
// ReqTypes.
const (
	ReqTypeSelect   = 101
	ReqTypeIndex    = 102
	ReqTypeDAG      = 103
	ReqTypeAnalyze  = 104
	ReqTypeChecksum = 105

	ReqSubTypeBasic      = 0
	ReqSubTypeDesc       = 10000
	ReqSubTypeGroupBy    = 10001
	ReqSubTypeTopN       = 10002
	ReqSubTypeSignature  = 10003
	ReqSubTypeAnalyzeIdx = 10004
	ReqSubTypeAnalyzeCol = 10005
)
```

As you can see, there are `ReqTypeDAG` and `ReqTypeAnalyze`, and also `ReqTypeChecksum` for `admin check table` ... for now, we just support `ReqTypeDAG`, so all other cases `req.Paging = false` is set.

A possible change is that we can move the `paging` property to `coprocessor.Request`.
That change would make it clear that `paging` is a property of `ReqTypeDAG` (or `coprocessor.Request` to be more specific), not a property `kv.Request`.
The drawback of the change is, if one day, we need paging protocol also for `Analyze/Checksum`, we need to add the property to `kv.Request`...

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**